src/Peachpie.CodeAnalysis/Symbols/Php/PhpRoutineSymbolExtensions.cs

```diff
@@ -144,6 +144,15 @@ public static TypeRefMask GetResultType(this IPhpValue symbol, TypeRefContext ct
                     Debug.Assert(t.IsSZArray());
                     return ctx.GetArrayTypeMask(TypeRefFactory.CreateMask(ctx, ((ArrayTypeSymbol)t).ElementType));
                 }
+                else if (ps.Syntax.TypeHint.IsCallable(out bool isNullable))
```

Comment:
`isNullable` is not correct, do not use it. Parameter can also be NULL is its default is `NULL`, use `ps.IsNotNull` instead